gemini/master.py

```diff
@@ -13,11 +15,48 @@
 NODE_TOKEN_BYTES = 32
 NODE_PASSWORD_BYTES = 32
 
+STORAGE_TYPE = "memory"
+
 app = Flask('gemini-master')
 
 c = consul.Consul(host='127.0.0.1', port=8500)
 
-jobs = []
+jobs: List[Any] = []
+
+
+class MemoryNodeInfoRepository:
+    def __init__(self):
+        self.node_data = {}
+
+    def put_data(self, value: dict):
+        self.node_data[value['node_id']] = value
+
+    def get_data(self, node_id: str) -> dict:
+        return self.node_data[node_id]
+
+
+class ConsulNodeInfoRepository:
+    def __init__(self):
+        self.node_data = {}
+
+    def put_data(self, value: dict):
+        c.kv.put('nodes/' + value['node_id'], json.dumps(value))
+
+    def get_data(self, node_id: str) -> dict:
+        self.node_data = c.kv.get('nodes/' + node_id)
+
+        if all(self.node_data):
+            output = json.loads(self.node_data[1]['Value'])
+
+        return output
+
+
+ds: Any
+
+if STORAGE_TYPE == "memory":
```

Comment:
Instead of resolving this in top level code. We should probably put this in a factory method and memoize it. You could use `@lru_cache(size=None)` for easy memoization/singleton.